File: leetcode腾讯50题打卡/Task_0/Ministack.py
class MinStack(object):

    def __init__(self):
        """
        initialize your data structure here.
        """
        self.data_stack = []
        self.mini_stack = []

    def push(self, x):
        """
        :type x: int
        :rtype: None
        """
        self.data_stack.append(x)
        if len(self.mini_stack)==0 or x <= self.mini_stack[-1]:
            self.mini_stack.append(x)
        

    def pop(self):
        """
        :rtype: None
        """
        if self.mini_stack[-1] == self.data_stack[-1]:
            self.mini_stack.pop()
        self.data_stack.pop()
        

    def top(self):
        """
        :rtype: int
        """
        return self.data_stack[-1]        

    def getMin(self):
        """
        :rtype: int
        """
        return self.mini_stack[-1]

File: leetcode腾讯50题打卡/Task_0/Ministack.cpp
#include<iostream>
#include<stack>

using namespace std;

class MinStack{
    public:
        stack<int> data_stack;
        stack<int> mini_stack;
        
        void push(int x){
            data_stack.push(x);
            if (mini_stack.empty()){
                mini_stack.push(x);
            }
            else if (mini_stack.top() >= x) {
                mini_stack.push(x);
            }
        }
    
        void pop(){
            if(mini_stack.top() == data_stack.top()){
                mini_stack.pop();
            }
            data_stack.pop();
        }
    
        int top(){
            return data_stack.top();
        }
    
        int getMin(){
            return mini_stack.top();        
        }
};

